src/main/java/com/boomi/flow/services/boomi/mdh/records/GoldenRecordRepository.java

```diff
@@ -176,6 +176,17 @@ public void delete(ApplicationConfiguration configuration, String universeId, Li
         return update(configuration, objects, universeId, null);
     }
 
+    private String getSourceIdFromProperty(Property property) {
+        if (GoldenRecordConstants.SOURCE_ID_FIELD.equals(property.getDeveloperName())) {
+            if (property.getContentValue() == null || property.getContentValue().isEmpty()) {
+                return GoldenRecordConstants.DEFAULT_SOURCE_ID;
+            } else {
+                return property.getContentValue();
+            }
+        }
+
+        throw new RuntimeException("Unexpected error, we can only get source from the source property");
```

Comment:
I'm not sure I understand what this error means

Reply:
it shouldn't fail never unless there is a problem in the code that is sending this property